pkg/cluster/builder.go

```diff
@@ -125,7 +125,9 @@ func (b FileSpecBuilder) Build(clusterConfigURL string) (*Spec, error) {
 	}
 
 	releaseVersion := v1alpha1.EksaVersion(release.Version)
-	config.Cluster.Spec.EksaVersion = &releaseVersion
+	if config.Cluster.IsSelfManaged() || (config.Cluster.IsManaged() && config.Cluster.Spec.EksaVersion == nil) {
```

Comment:
Why is the logic different for management and workload clusters?

Sorry if I'm resurfacing something we talked about already, but I would love if we could avoid this complexity. It's not the code that concerns me, but the user experience.

This seems like some nuanced logic that we need to document and the user has to take into account on every upgrade. And although I'm sure you have a good reason to do this, I don't thin it's very apparent, which in my experience ends up with the user forgetting.

Could we just either always overwrite it or never overwrite if already set?

Reply:
Yes, we can continue to always overwrite it.